tests/kernel/kernel_constructors.cpp

```diff
@@ -162,109 +157,121 @@ class TEST_NAME : public sycl_cts::util::test_base {
         cts_selector ctsSelector;
         auto ctsQueue = util::get_cts_object::queue(ctsSelector);
         auto deviceList = ctsQueue.get_context().get_devices();
-
-        if (!is_compiler_available(deviceList)) {
-          log.note(
-              "online compiler is not available -- skipping test of equality "
-              "operator");
-        } else {
-          sycl::program prog(ctsQueue.get_context());
-          prog.build_with_kernel_type<test_kernel<4>>();
-          auto kernelA = prog.get_kernel<test_kernel<4>>();
-          ctsQueue.submit([&](sycl::handler &cgh) {
-            cgh.single_task(test_kernel<4>());
-          });
-
-          sycl::kernel kernelB(kernelA);
-
-          sycl::program progC(ctsQueue.get_context());
-          progC.build_with_kernel_type<test_kernel<5>>();
-          auto kernelC = progC.get_kernel<test_kernel<5>>();
-          ctsQueue.submit([&](sycl::handler &cgh) {
-            cgh.single_task(test_kernel<5>());
-          });
-          kernelC = (kernelA);
-
-          sycl::program progD(ctsQueue.get_context());
-          progD.build_with_kernel_type<test_kernel<6>>();
-          auto kernelD = progD.get_kernel<test_kernel<6>>();
-          ctsQueue.submit([&](sycl::handler &cgh) {
-            cgh.single_task(test_kernel<6>());
-          });
-
-          if (!ctsSelector.is_host()) {
-#ifdef SYCL_CTS_TEST_OPENCL_INTEROP
-            if (kernelA == kernelB &&
-                (kernelA.get() != kernelB.get() ||
-                 kernelA.get_context().get() != kernelB.get_context().get() ||
-                 kernelA.get_program().get() != kernelB.get_program().get())) {
-              FAIL(
-                  log,
-                  "kernel equality does not work correctly (copy constructed)");
+        auto ctx = ctsQueue.get_context();
+
+        using k_name4 = test_kernel<4>;
+        auto kbA = sycl::get_kernel_bundle<
+                      k_name4, sycl::bundle_state::executable>(ctx);
+        auto kernelA = kbA.get_kernel(sycl::get_kernel_id<k_name4>());
+        ctsQueue.submit([&](sycl::handler &cgh) {
+          cgh.single_task(k_name4());
+        });
+        sycl::kernel kernelB(kernelA);
+
+        using k_name5 = test_kernel<5>;
+        auto kbC = sycl::get_kernel_bundle<
+                      k_name5, sycl::bundle_state::executable>(ctx);
+        auto kernelC = kbC.get_kernel(sycl::get_kernel_id<k_name5>());
+        ctsQueue.submit([&](sycl::handler &cgh) {
+          cgh.single_task(k_name5());
+        });
+        kernelC = (kernelA);
+
+        using k_name6 = test_kernel<6>;
+        auto kbD = sycl::get_kernel_bundle<
+                      k_name6, sycl::bundle_state::executable>(ctx);
+        auto kernelD = kbD.get_kernel(sycl::get_kernel_id<k_name6>());
```

Comment:
@gmlueck do you think it is interesting to provide in the specification a short-cut for this pattern?

Reply:
Yes, I think that would be easy to do.  If we do this, we should add similar overloads for `has_kernel`.  I think the implementation would be like:

```
class kernel_bundle {
 public:
  template<KernelName>
  bool has_kernel() { return has_kernel(get_kernel_id<KernelName>()); }

  template<KernelName>
  bool has_kernel(const device &dev) { return has_kernel(get_kernel_id<KernelName>(), dev); }

  template<KernelName>
  bool get_kernel() { return get_kernel(get_kernel_id<KernelName>()); }
};
```

@bader: Do you see any reason why this would cause problems?

Reply:
> @bader: Do you see any reason why this would cause problems?

No. :)

Reply:
Shortcuts added to the spec in KhronosGroup/SYCL-Docs#194